Shared/ThresholdKeyView.swift

```diff
@@ -108,6 +110,11 @@ struct ThresholdKeyView: View {
                                     showAlert = true
                                     tkeyReconstructed = true
                                     resetAccount = false
+                                    let shareIndexes = try threshold_key.get_shares_indexes()
+                                    let share = try! threshold_key.output_share(shareIndex: shareIndexes[1], shareType: nil)
+                                    let saveId = fetchKey + ":1" //save the device share
+                                    print(shareIndexes)
+                                    try! KeychainInterface.save(item: share, key: saveId)
```

Comment:
This is already being done further down in this function
```suggestion
```

Reply:
I think we need to save device share in this step.
If we don't save here and login next time,
since the account is existing, passes the saving logic at line 186.

Reply:
The comparison is against locally saved shares.

If you haven't saved any shares, you're assumed to be a new user, if you have lost shares reconstruction will fail.

If you have already saved shares, they will be loaded and reconstruction would pass.

If you're a new user, the shares will be saved, if you open the app again with the persisted shares, you'll land up at the previous statement.

Reply:
So you mean the share will be saved without that saving logic, if I'm a new user and initialize new key?
I'm afraid that's wrong.

Assume that we don't the saving logic at line 127~141
```
//                                    let shareIndexes = try threshold_key.get_shares_indexes()
//
//                                    // let's get the device share index
//                                    var deviceShareIndex = ""
//                                    // "1" is for the social login share
//                                    if shareIndexes[0] == "1" {
//                                        deviceShareIndex = shareIndexes[1]
//                                    } else {
//                                        deviceShareIndex = shareIndexes[0]
//                                    }
//
//                                    let share = try threshold_key.output_share(shareIndex: deviceShareIndex, shareType: nil)
//                                    let saveId = fetchKey + ":0"
//                                    //save the device share
//                                    try KeychainInterface.save(item: share, key: saveId)
```
And try clicking `Initialize and reconstruct tkey` 2 times in a row, with a new user.

First one will succeed, but second one will go to fail, with error `Failed to reconstruct key with available shares`
it's because we can't import any shares, since we don't have any in local.

That's why I think saving logic should be included for the new user case.
Try this edge case, and tell me if there's some better way to solve this.

Reply:
Hmm, I'll check it again. 

It's likely a bug that was missed in a PR, not an edge case, since reconstruction should not be happening above this line in master:
```
     // fetch all locally available shares for this google account
```

Only this part needed to be adapted to save a single share, instead of all of them.
```
// fetch all locally available shares for this google account
                                var shares: [String] = []
                                shareCount = 0
                                var finishedFetch = false
                                repeat {
                                    let fetchId = fetchKey + ":" + String(shareCount)
                                    do {
                                        let share = try KeychainInterface.fetch(key: fetchId)
                                        shares.append(share)
                                    } catch {
                                        finishedFetch = true
                                        break
                                    }
                                    shareCount += 1
                                } while !finishedFetch

                                // There are 0 locally available shares for this tkey
                                if shareCount == 0 {
                                    // Attempt reconstruction
                                    guard let reconstructionDetails = try? await threshold_key.reconstruct() else {
                                        alertContent = "Failed to reconstruct key. \(threshold) more share(s) required"
                                        resetAccount = true
                                        showAlert = true
                                        showSpinner = SpinnerLocation.nowhere
                                        return
                                    }

                                    // save shares up to required threshold
                                    let shareIndexes = try threshold_key.get_shares_indexes()

                                    // TODO: Save only one share, which is device share and not all shares.
                                    for i in 0..<threshold {
                                        let saveId = fetchKey + ":" + String(i)

                                        guard let share = try? thresholdKey.output_share(shareIndex: shareIndexes[Int(i)], shareType: nil) else {
                                            alertContent = "Failed to output share"
                                            resetAccount = true
                                            showAlert = true
                                            showSpinner = SpinnerLocation.nowhere
                                            return
                                        }

                                        guard let _ = try? KeychainInterface.save(item: share, key: saveId) else {
                                            alertContent = "Failed to save share"
                                            resetAccount = true
                                            showAlert = true
                                            showSpinner = SpinnerLocation.nowhere
                                            return
                                        }
                                    }

                                    reconstructedKey = reconstructionDetails.key
                                    alertContent = "\(reconstructedKey) is the private key"
                                    showAlert = true
                                    tkeyReconstructed = true
                                    resetAccount = false
                                    showSpinner = SpinnerLocation.nowhere
                                }
                                // existing account
                                else {
                                    // check enough shares are available
                                    if shareCount < threshold {
                                        alertContent = "Not enough shares for reconstruction"
                                        showAlert = true
                                        resetAccount = true
                                        showSpinner = SpinnerLocation.nowhere
                                        return
                                    }
                                    // import shares
                                    // TODO: Handle failures of input_share gracefully. Its possible that locally available share was deleted.
                                    for item in shares {
                                        guard let _ = try? await threshold_key.input_share(share: item, shareType: nil) else {
                                            alertContent = "Incorrect share was used"
                                            showAlert = true
                                            resetAccount = true
                                            showSpinner = SpinnerLocation.nowhere
                                            return
                                        }
                                    }

                                    guard let reconstructionDetails = try? await threshold_key.reconstruct() else {
                                        alertContent = "Failed to reconstruct key with available shares."
                                        resetAccount = true
                                        showAlert = true
                                        showSpinner = SpinnerLocation.nowhere
                                        return
                                    }

                                    reconstructedKey = reconstructionDetails.key
                                    alertContent = "\(reconstructedKey) is the private key"
                                    showAlert = true
                                    tkeyReconstructed = true
                                    resetAccount = false
                                }
                                showSpinner = SpinnerLocation.nowhere
                            }
```

Reply:
Please rebase onto this branch and check if your previous comment still holds true.
https://github.com/torusresearch/tkey-rust-ios-example/pull/50

Reply:
rebased and works perfect.